p2p/net/network.go

```diff
@@ -285,6 +285,15 @@ func (n *Net) Shutdown() {
 	}
 }
 
+func (n *Net) isShuttingDown() bool {
```

Comment:
we could probably refactor this to a general `IsShuttingDown(context)` but this is really matter of preference since this is not a lot of code to duplicate, just a lot of places, if we ever add changes to this method it's worth extracting it out.

Reply:
i was more worried about circular dep (where to put this util func, in p2p/ or in its own package.
a util package for this seems overkill. since i will continue working on this package for the Tomb package, i prefer to defer this till later changes.

Reply:
eventually will need its own package inside p2p/whatever or util in order to prevent the circular dep..